### PR TITLE
Bp remove codesmells

### PR DESCRIPTION
**Pull request by Johannes-Schneider**

### CODE REVIEW

File: src/main/java/de/hpi/bpt/argos/persistence/database/DatabaseConnectionImpl.java [hunk 1]
```diff
@@ -472,6 +303,41 @@ public void deleteEntities(PersistenceEntity... entities) {
...
+								   Query<Q> query,
+								   Transaction transaction,
+								   Callable<R> getValue,
+								   R defaultValue) {
+
+		try {
+			R result = getValue.call();
+			transaction.commit();
+			return result;
+		} catch (NoResultException e) {
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/persistence/database/DatabaseConnectionImpl.java]: MINOR Either log or rethrow this exception. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/main/java/de/hpi/bpt/argos/persistence/database/DatabaseConnectionImpl.java (hunk 1 above)
Comment (tommartensen) [inline src/main/java/de/hpi/bpt/argos/persistence/database/DatabaseConnectionImpl.java:322]: @Hannes01071995 please log this exception (I know you don't like this, but sonarqube sees this as a bug...

Comment (tommartensen) [reply src/main/java/de/hpi/bpt/argos/persistence/database/DatabaseConnectionImpl.java:322]: and the breaks in the signature look strange, maybe there is a better way to break them

**Review by tommartensen — CHANGES_REQUESTED** [review]

**Pull request merged**